guarddog/utils/archives.py

```diff
@@ -1,49 +1,34 @@
 import logging
 import os
-import zipfile
 import stat
+import zipfile
+
 import tarsafe  # type:ignore
 
 log = logging.getLogger("guarddog")
 
 
 def is_supported_archive(path: str) -> bool:
     """
-    Decide whether a file contains a supported archive.
+    Decide whether a file contains a supported archive based on its
+    file extension.
 
     Args:
         path (str): The local filesystem path to examine
 
     Returns:
         bool: Represents the decision reached for the file
-    """
-    return is_tar_archive(path) or is_zip_archive(path)
-
 
-def is_tar_archive(path: str) -> bool:
     """
-    Decide whether a file contains a tar archive.
+    def is_tar_archive(path: str) -> bool:
+        tar_exts = [".bz2", ".bzip2", ".gz", ".gzip", ".tgz", ".xz"]
```

Comment:
can't we use [tarfile.is_tarfile](https://docs.python.org/3/library/tarfile.html#tarfile.is_tarfile)?

Reply:
So it turns out that this function `is_supported_archive()` is only used once, in `PypiPackageScanner.download_package()`.  GuardDog uses it to filter distribution files based on the file extension before actually committing to downloading one.

We could change the behavior so that it always download all the files and then use `tarsafe.is_tarfile` to select one, but I think it's advantageous to keep the current behavior and save on potentially unnecessary downloads.